plugin/eqdetect/source/dextool/plugin/eqdetect/runner.d

```diff
@@ -57,9 +57,12 @@ ExitStatusType runPlugin(string[] args) {
 
         exit_status = analyzeFile(AbsolutePath(FileName(m.path)), cflags, visitor, ctx);
         import std.path : baseName;
-        import dextool.plugin.eqdetect.subfolder : writeToFile;
-        writeToFile(visitor.generatedSource.render, baseName(m.path), m.kind, m.id, false);
-        writeToFile(visitor.generatedMutation.render, baseName(m.path), m.kind, m.id, true);
+        import dextool.plugin.eqdetect.subfolder : writeToFile, SnippetFinder;
+        string source_path = writeToFile(visitor.generatedSource.render, baseName(m.path), m.kind, m.id, "_source_");
```

Comment:
if it is a path I would recommend to use the struct AbsolutePath or variants of it. It ensures that the path is **always** absolute. It makes it easier to pass them around in the program.

`source/dextool/type.d`

Reply:
it also avoid extra work such as at other places in the program checking if it is a path. You can see it from the opAssign overloads and constructors.

In this case it could be written as:
`auto source_p = ....Path.AbsolutePath`